test/integration/targets/lookups/template_deepcopy/playbook.yml

```diff
@@ -0,0 +1,10 @@
+- hosts: h1
+  gather_facts: no
+  tasks:
+    - set_fact:
+        templated_foo: "{{ lookup('template', 'template.in') }}"
+
+    - name: Test that the hostvar was templated correctly
+      assert:
+        that:
+          - templated_foo == "foo\n"
```

Comment:
What would be the gain? In tests I don't really care about duplicate code. I prefer to use the actual values for comparisons. That way it does not depend on the templating engine (like `{{hostvars['h1'].host_var}}` might equal to `{{ host_var }}` but if it is templated to an incorrect value we won't find out because the comparison would still be true, however it's unlikely).

Reply:
Given it's the actual templating engine that's being tested I guess hard coded values makes sense.  I would have switched to the variable just because folks looking to write their first tests often look for simple examples and assume the simple example is the way it should always be done.